Prog/Eviction Group/Pulse-housing-analysis.R

```diff
@@ -0,0 +1,192 @@
+# Author: Elizabeth Burton 
+# Created: 1/17/24
+# Description: compile, clean, and analyze PUF census data for DC housing assistance analysis
+######
+
+library(tidyverse)
+library(readxl)
+library(openxlsx)
+library(data.table)
+library(srvyr)
+library(weights)
+
+# Read in data and replicate weights: using week 57-week 63 surveys (May 2023-Oct 2023) and cycle 01 (Jan-Feb 2024)
+## list.files() extracts the names of all .csv files at the specified path. 
+## map_dfr() and read_csv read in each .csv file and combine them all into a single dataset, called data. 
+## Resulting data has 629,582 rows and 283 columns, and rep_weights has 823,537 rows and 163 columns.
+data <- list.files(path="//sas1/dcdata/Libraries/Requests/Raw/Eviction Group/", pattern = "[.]csv$", full.names=TRUE ) %>% 
+  map_dfr(read_csv)
+
+# Create categorical variables 
+clean_data <- data %>%
+  filter(EST_ST == "11", # DC only
+         TENURE == 3 | TENURE == 4 ) %>% # renters only 3 == Rented; 4 == Occupied without payment of rent
+  mutate(pressured = ifelse(MOVEWHY1 == 1|MOVEWHY2 == 1| MOVEWHY3 == 1|MOVEWHY4 == 1|MOVEWHY5 == 1|MOVEWHY6 == 1|MOVEWHY7 == 1,
+                              "Pressure", # Respondents can choose multiple reasons
+                              ifelse(MOVEWHY8 == 1, "No pressure", "Not reported")),
+           increase_rent = case_when(MOVEWHY1 == 1 ~ "Increased rent"), 
+           missed_rent = case_when(MOVEWHY2 == 1 ~ "Missed Rent"), 
+           repairs_not_made = case_when(MOVEWHY3 == 1 ~ "Repairs not made"),
+           eviction_threatened = case_when(MOVEWHY4 == 1 ~ "Eviction threatened"),
+           locks_changed = case_when(MOVEWHY5 == 1 ~ "Locks changed"),
+           nhbd_danger = case_when(MOVEWHY6 == 1 ~ "Neighborhood dangerous"),
+           other_pressure = case_when(MOVEWHY7 == 1 ~ "Other pressure"),
+           moved = case_when(MOVED == 1 ~ "Moved from Pressure", 
+                             MOVED == 2 ~ "Did not move from Pressure",
+                             MOVED == -99 | MOVED == -88 ~ "not reported"),
+           income = case_when(INCOME == 1 ~ "$25,000 below",  
+                            INCOME == 2 | INCOME == 3 ~ "$25,000-$49,999",
+                            INCOME == 4 ~ "$50,000-$74,999",
+                            INCOME == 5 ~ "$75,000-$99,999",
+                            INCOME == 6 | INCOME == 7 | INCOME == 8 ~ "$100,000 above",
+                            INCOME == -99 | INCOME == -88 ~ "not reported"),
+           rent_behind = case_when(RENTCUR == 1 ~ "Not behind on rent",
+                                   RENTCUR == 2 ~ "Behind on rent",
+                                   RENTCUR == -99 | RENTCUR == -88 ~ "not reported"),
+           behind_two_months = case_when(TMNTHSBHND >= 2 ~"Behind 2+ months rent", 
+                                         TMNTHSBHND == -99 | TMNTHSBHND == -88 ~ "not reported",
+                                         TRUE ~ "Not behind on rent"),
```

Comment:
I assume "not behind on rent" is actually not 2 months behind on rent? can you update?